bundles/org.openhab.binding.airgradient/src/main/java/org/openhab/binding/airgradient/internal/handler/AirGradientAPIHandler.java

```diff
@@ -0,0 +1,401 @@
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpMethod;
+import org.openhab.binding.airgradient.internal.config.AirGradientAPIConfiguration;
+import org.openhab.binding.airgradient.internal.discovery.AirGradientLocationDiscoveryService;
+import org.openhab.binding.airgradient.internal.model.LedMode;
+import org.openhab.binding.airgradient.internal.model.Measure;
+import org.openhab.binding.airgradient.internal.prometheus.PrometheusMetric;
+import org.openhab.binding.airgradient.internal.prometheus.PrometheusTextParser;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * The {@link AirGradientAPIHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jørgen Austvik - Initial contribution
+ */
+@NonNullByDefault
+public class AirGradientAPIHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(AirGradientAPIHandler.class);
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private final HttpClient httpClient;
+    private final Gson gson;
+
+    public AirGradientAPIHandler(Bridge bridge, HttpClient httpClient) {
+        super(bridge);
+        this.httpClient = httpClient;
+        this.gson = new Gson();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.info("Handle command {} for channel {}!", command.toFullString(), channelUID);
+        if (command instanceof RefreshType) {
+            pollingCode();
+        } else {
+            // This is read only
+            logger.debug("Received command {} for channel {}, but the API is read only", command.toString(),
```

Comment:
If i understand this correct, this would be a read-only value and writing to this would be weird. So i guess something is wrong and the user should be aware of this by a warn logger. Does that make sense?
```suggestion
            logger.warn("Received command {} for channel {}, but the API is read only", command.toString(),
```